src/engine/TransitivePath.cpp

```diff
@@ -726,8 +726,12 @@ std::shared_ptr<TransitivePath> TransitivePath::bindLeftSide(
     std::shared_ptr<QueryExecutionTree> leftop, size_t inputCol) const {
   // Enforce required sorting of `leftop`.
   leftop = QueryExecutionTree::createSortedTree(std::move(leftop), {inputCol});
-  // Create a copy of this
-  std::shared_ptr<TransitivePath> p = std::make_shared<TransitivePath>(*this);
+  // Create a copy of this. Important: make sure that the variable column map is
+  // computed before the copy is made.
```

Comment:
The problem is rather something like this:
```
Note: We cannot simply use the copy constructor, because then the base class `Operation` would ignore the changes in the `variableColumnMap_` that are made below (For details see `Operation::getInternallyVisibleVariableColumns` and `Operation::getExternallyVariableColumns`.
```
(Same comment for the `bindRightSide` variation below).